<launch>

    <arg name="is_sim" default="false"/>


    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(find zetaxbot_controller)/config/trajectory_controllers.yaml" command="load"/>

    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(find zetaxbot_controller)/config/joint_state_controller.yaml" command="load"/>

    <param name="robot_description" command="$(find xacro)/xacro $(find zetax)/urdf/zetax.urdf.xacro"/>

    <!-- If interfacing with the real Hardware -->
    <group unless="$(arg is_sim)">
        <!-- Launch the angles converter service -->
        <node pkg="zetaxbot_controller" type="angles_converter.py" name="angles_converter" 
            respawn="true" output="screen"/>

        <!-- Launch the Hardware interface with the real robot -->
        <node name="zetaxbot_interface_node" pkg="zetaxbot_controller" type="zetaxbot_interface_node" 
            ns="/zetaxbot" respawn="true" output="screen"/>
    </group>

    <!-- Load the controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" 
        ns="/zetaxbot" args="joint_state_controller right_arm_controller left_arm_controller head_controller" />

    <!-- Convert joint states to TF transforms for rviz, etc -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
        <remap from="/joint_states" to="/zetaxbot/joint_states" />
    </node>

</launch>